src/Components/BranchesControl.jsx

```diff
@@ -0,0 +1,144 @@
+import React, {useContext, useEffect, useState} from 'react'
+import {useNavigate} from 'react-router-dom'
+import TextField from '@mui/material/TextField'
+import MenuItem from '@mui/material/MenuItem'
+import Paper from '@mui/material/Paper'
+import Typography from '@mui/material/Typography'
+import {makeStyles} from '@mui/styles'
+import {ColorModeContext} from '../Context/ColorMode'
+import debug from '../utils/debug'
+import {getBranches} from '../utils/GitHub'
+import useStore from '../store/useStore'
+import {navigateBaseOnModelPath} from '../utils/location'
+
+
+/**
+ * @param {boolean} isDialogDisplayed
+ * @param {Function} setIsDialogDisplayed
+ * @return {object} React component
+ */
+export default function Branches() {
+  const classes = useStyles()
+  const navigate = useNavigate()
+  const colorMode = useContext(ColorModeContext)
+  const repository = useStore((state) => state.repository)
+  const [branches, setBranches] = useState([])
+  const [versionPaths, setVersionPaths] = useState([])
+  const [selected, setSelected] = useState(0)
+  const modelPath = useStore((state) => state.modelPath)
+
+  useEffect(() => {
+    if (!repository) {
+      debug().warn('IssuesControl#Issues: 1, no repo defined')
+      return
+    }
+    const fetchBranches = async () => {
+      try {
+        const branchesData = await getBranches(repository)
+        const versionPathsTemp = []
+        if (branchesData.data.length > 0) {
+          setBranches(branchesData.data)
+        }
+        branchesData.data.map((branch, i) => {
+          if (branch.name === modelPath.branch) {
+            // select the current branch
+            setSelected(i)
+          }
+          const versionPath = navigateBaseOnModelPath(modelPath.org, modelPath.repo, branch.name, modelPath.filepath)
+          versionPathsTemp.push(versionPath)
+        })
+        setVersionPaths(versionPathsTemp)
+      } catch (e) {
+        debug().warn('failed to fetch branches', e)
+      }
+    }
+
+    if (branches.length === 0 && modelPath.repo !== undefined) {
+      fetchBranches()
```

Comment:
Does await on this break the test?  If not I think the way we had it with the IIFE is more correct

Reply:
I don't think we need IIFE - i have never seen this pattern, and the logic works.